addons/zylann.editor_debugger/dock.gd

```diff
@@ -317,25 +337,45 @@ static func restore_ownership(root: Node, owners: Dictionary, include_internal:
 		restore_ownership(child, owners, include_internal)
 
 
-func _on_ShowInInspectorCheckbox_toggled(button_pressed):
+func _on_ShowInInspectorCheckbox_toggled(_button_pressed: bool) -> void:
 	pass
 
 
-func _on_SaveBranchAsSceneButton_pressed():
-	#_save_branch_as_scene_button.accept_event()
+func _on_popup_menu_index_pressed(index: int) -> void:
 	_popup_menu.hide()
-	_save_branch_file_dialog.popup_centered_ratio()
-
-
-func _on_SaveBranchFileDialog_file_selected(path: String):
-	var node_view = _tree_view.get_selected()
-	var node = _get_node_from_view(node_view)
+	match index:
+		POPUP_ACTIONS.SAVE_BRANCH_AS_SCENE:
+			_save_branch_file_dialog.popup_centered_ratio()
+		POPUP_ACTIONS.COPY_PATH_TO_CLIPBOARD:
+			var node_view := _tree_view.get_selected()
+			var node := _get_node_from_view(node_view)
+			DisplayServer.clipboard_set(node.get_path())
+			print("Copied to clipboard: %s"%[node.get_path()])
+		POPUP_ACTIONS.COPY_NODE_TYPES_TO_CLIPBOARD:
```

Comment:
I agree with you on points 1 and 2, but not 3; The nodes can certainly be queried by path, but it's a bit of annoying additional work on someone who wants to use the plugin specifically to build up their own.

I think "copy typed path" with just the classes can be ok, and solves your points 1 and 2. It does make the function less useful though, I think.

I personally don't have a need for references; I want to build plugins that hook into the editor's interface, so I need to be able to construct functions that find those nodes the most efficiently possible.

"copy path" by itself is rarely useful, because the path is almost never guaranteed. But "copy types" is also not fully useful; Often, class names are sufficient to find a specific node. Sometimes though, you can discriminate better by using a name; some nodes have a consistent name.

Having both "copy path" and "copy types" requires to click the pop-up twice, then paste twice, and then visually match a path chunk with a type to get a clear idea of what's the best query to write to get a specific node.

I did that a bunch of times, then decided on this structure because it allowed me to parse it (visually) much faster.

I think this is a feature that's useful to anyone using this plugin (since I suppose the main usage of this plugin is to find path to native UI elements in the editor?). Unless I misunderstand the primary usage people use this for.

Having the name and class next to each other in a structured format even potentially allows for generating a query function automatically ("if node name does not start with `@`, use that because it's unique, ottherwise use the class name").

This said, if it's just useful to me, I'll squish it and reimplement this in a personal tool.

Reply:
> I think this is a feature that's useful to anyone using this plugin (since I suppose the main usage of this plugin is to find path to native UI elements in the editor?

This is a valid use case for the option. It's not something one should rely on too much (since it's reverse-engineering and could break anytime the devs change structure), but usually relies on guessing the access to it based on a typed path that can be given to a helper function.
I just prefer if the menu is named accordingly because the initial name wasn't explicit enough IMO.

Reply:
How about also adding tooltips?:

```gdscript
	POPUP_ACTIONS.SAVE_BRANCH_AS_SCENE: {
		"title": "Save branch as scene",
		"tooltip": "Save the branch as a new scene in a directory of your choice"
	},
	POPUP_ACTIONS.COPY_PATH_TO_CLIPBOARD: {
		"title": "Copy path to clipboard",
		"tooltip": "Copy the path to the node in the format \"/path/to/node\""
	},
	POPUP_ACTIONS.COPY_NODE_TYPES_TO_CLIPBOARD:{
		"title": "Copy typed path to clipboard",
		"tooltip": "Copy the path to the node in the format [[\"type\", \"node\"], [\"type\", \"node\"], ...]"
	},
```

Reply:
Example of generated function:

```gdscript
	match ... snip ...
		POPUP_ACTIONS.COPY_FUNCTION_TO_CLIPBOARD:
			var node_view := _tree_view.get_selected()
			var node := _get_node_from_view(node_view)
			var created_func_name := "get_" + node.get_class() + "_" + node.name.replace("@","")
			var func_lines := []
			while node.get_parent():
				var parent := node.get_parent()
				var line := "\n".join([
					"\tfor child in node.get_children():",
					"\t\tif child.%s == \"%s\":"%(["get_class()", node.get_class()] if node.name.begins_with("@") else ["name", node.name]),
					"\t\t\tnode = child",
					"\t\t\tbreak" if parent != null else "\t\t\treturn child",
					"",
				])
				func_lines.push_front(line)
				node = parent
			var function := "".join([
				"func %s(node: Node = get_tree().root) -> Node:\n"%[created_func_name],
				"".join(func_lines),
				"\treturn null"
			]) 
			DisplayServer.clipboard_set(function)
			print("Copied to clipboard: %s"%[function])
```

Reply:
Tooltips sounds good.

I thought of generating code too but when used often it makes for quite redundant and verbose code, so not sure if it's better than having a helper to which can be sent the typed path, considering all the checks that each node could require.

Reply:
See if this works for you?